scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -198,6 +199,68 @@ def test_shortest_path_min_only_random(n):
             p = pred[p]
 
 
+# ensure dijkstra does not crash or run into an infinite loop
+@pytest.mark.parametrize('n, t',
+                         ((3, 300),
+                          (10, 100),
+                          (100, 10),
+                          (500, 1)))
+def test_dijkstra_random(n, t):
+    np.random.seed(1234)
+    for testcase_index in range(t):
```

Comment:
I think we can avoid iterating over `t` to call `dijkstra()` over and over here--in fact, I couldn't quite figure out why you were doing that with a fixed seed, but it looks like the seed may not have been passed to `rand`, so you were doing a random search